book/src/SUMMARY.md

```diff
@@ -2,6 +2,7 @@
 
 - [failure](./intro.md)
 - [How to use failure](./howto.md)
+	- [bail! and ensure!](./bail-and-ensure.md)
```

Comment:
Could you move this to the bottom of this section, instead of the top? I think the core mechanisms of failure should be learned first before reading about these macros.